pkg/eks/client.go

```diff
@@ -85,6 +80,50 @@ func (c *Client) useEmbeddedToken(spec *api.ClusterConfig, stsclient stsiface.ST
 	return nil
 }
 
+type generator struct{}
+
+// Token is generated and used by Kubernetes client-go to authenticate with a Kubernetes cluster.
+type Token struct {
+	Token      string
+	Expiration time.Time
+}
+
+const (
+	clusterIDHeader        = "x-k8s-aws-id"
+	presignedURLExpiration = 15 * time.Minute
+	v1Prefix               = "k8s-aws-v1."
+)
+
+// GetWithSTS returns a token valid for clusterID using the given STS client.
+// This implementation follows the steps outlined here:
+// https://github.com/kubernetes-sigs/aws-iam-authenticator#api-authorization-from-outside-a-cluster
+// We either add this implementation or have to maintain two versions of STS since aws-iam-authenticator is
+// not switching over to aws-go-sdk-v2.
+func (g generator) GetWithSTS(ctx context.Context, clusterID string, presigner *sts.PresignClient) (Token, error) {
```

Comment:
Sadly, the PresignGetCallerIdentity is on the client and not on any interfaces. And so, we require the real thing here.